realm/realm-library/src/main/java/io/realm/HandlerController.java

```diff
@@ -469,13 +466,11 @@ private void completedAsyncQueriesUpdate(QueryUpdateTask.Result result) {
                 query.notifyChangeListeners();
             }
 
-            // notify listeners only when we advanced
-            if (compare != 0) {
-                notifyGlobalListeners();
-                // notify RealmResults & RealmObject callbacks (type based notifications)
-                notifySyncRealmResultsCallbacks();
-                notifyRealmObjectCallbacks();
-            }
+            // We need to notify the rest of listeners, since the original REALM_CHANGE
+            // was delayed/swallowed in order to be able to update async queries
```

Comment:
Like I mentioned in the comment I've added. This `updateAsyncEmptyRealmObject` is a consequence of a `REALM_CHANGE` we delayed/swallowed, so even if we don't advance the Realm, we still need to update sync RealmResults & RealmObject (because we originally delayed their notif  until this completes)